From zaitcev@redhat.com  Mon Feb 12 22:56:30 2007
From: Pete Zaitcev <zaitcev@redhat.com>
Date: Sun, 11 Feb 2007 13:56:13 -0800
Subject: USB: make usb_iso_packet_descriptor.status signed
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net
Message-ID: <20070211135613.99acaf76.zaitcev@redhat.com>


The status in usb_iso_packet_descriptor should be signed, for the benefit
of someone who casts to a long or makes other benign misstep (the principle
of least surprise).

Signed-off-by: Pete Zaitcev <zaitcev@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/usb.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/include/linux/usb.h
+++ gregkh-2.6/include/linux/usb.h
@@ -935,7 +935,7 @@ struct usb_iso_packet_descriptor {
 	unsigned int offset;
 	unsigned int length;		/* expected length */
 	unsigned int actual_length;
-	unsigned int status;
+	int status;
 };
 
 struct urb;
